### PR TITLE
Disabling filtering btn on invalid input

### PR DESCRIPTION
**Pull request by zaelgohary**
Related issue: https://github.com/threefoldtech/tfgrid_dashboard/issues/170

### CODE REVIEW

**Review by rawdaGastan — COMMENTED** [review]
What about nodes case? the filters don't have a button to be disabled

**Comment by rawdaGastan** [comment]
I think in the farms page if we added a new filter after displaying the error and make the button disabled, it will be enabled again

**Comment by zaelgohary** [comment]
> What about nodes case? the filters don't have a button to be disabled

Filters in the nodes page work differently than the ones in the farms. You can now only filter with one input & if multiple inputs were added, the last one would only work. so, no need to add extra validation.

[nodesFilter.webm](https://user-images.githubusercontent.com/40770501/200165594-4595f20b-87a3-4fe6-a742-84830cac6506.webm)

**Comment by zaelgohary** [comment]
> What about nodes case? the filters don't have a button to be disabled

disabled filtering in case an error is displayed.

**Comment by zaelgohary** [comment]
> I think in the farms page if we added a new filter after displaying the error and make the button disabled, it will be enabled again

I tried to separate the invalid value for each filter in order to prevent this, also tried to change the invalid value when the filter gets destroyed but the deleted filter always returned invalid in case it had an error before deletion.

File: src/explorer/views/Farms.vue [hunk 1]
```diff
@@ -349,5 +354,17 @@ export default class Farms extends Vue {
...
+
+  invalidFilter(event: {symbol: string, invalid: boolean}) {
+    const checkFilter = this.activeFilters.find((f) => (f.symbol === event.symbol));
+    if(checkFilter){
+      checkFilter.invalid = event.invalid;
+    }
+  }
+
+  isInvalid():boolean {
+    console.log("activefilters.some", this.activeFilters.map((f) => f.invalid));
```
Comment (rawdaGastan) [inline src/explorer/views/Farms.vue]: can we remove this? It logs whenever we change the filters

File: src/explorer/views/Farms.vue [hunk 2]
```diff
@@ -129,11 +131,14 @@ import LayoutFilters from "../components/LayoutFilters.vue";
   },
 })
 export default class Farms extends Vue {
+  log = console.log.bind(console);
```
Comment (rawdaGastan) [inline src/explorer/views/Farms.vue]: I don't think we need it. What do you think?

File: src/explorer/components/NodeFilter.vue [hunk 3]
```diff
@@ -62,15 +64,9 @@ export default class InFilter extends Vue {
       key: this.filterKey,
       value,
     });
-
-    // reset to the first page
-    this.$store.commit(
```
Comment (rawdaGastan) [inline src/explorer/components/NodeFilter.vue:67]: why did you remove this?

File: src/explorer/components/NodeFilter.vue [hunk 4]
```diff
@@ -62,15 +64,9 @@ export default class InFilter extends Vue {
...
-      "explorer/" + MutationTypes.SET_NODES_TABLE_PAGE_NUMBER,
-      1
-    );
-
-    // load nodes with the changes
-    this.$store.dispatch(ActionTypes.REQUEST_NODES);
+    if(!this.invalid){
```
Comment (rawdaGastan) [inline src/explorer/components/NodeFilter.vue]: Can we make the condition at the top of the function? Or you Have a different opinion?

**Review by rawdaGastan — CHANGES_REQUESTED** [review]

File: src/explorer/components/NodeFilter.vue (hunk 3 above)
Comment (zaelgohary) [reply src/explorer/components/NodeFilter.vue:67]: It was removed by mistake.

File: src/explorer/components/NodeFilter.vue (hunk 4 above)
Comment (zaelgohary) [reply src/explorer/components/NodeFilter.vue]: sure.

File: src/explorer/views/Farms.vue (hunk 2 above)
Comment (zaelgohary) [reply src/explorer/views/Farms.vue]: done.

File: src/explorer/views/Farms.vue (hunk 1 above)
Comment (zaelgohary) [reply src/explorer/views/Farms.vue]: done.

**Review by rawdaGastan — APPROVED** [review]
